### ISSUE TITLE
[Warning] Cuda memory problem in Pipeline._val_loop

### ISSUE THREAD

**Issue by raphaelreinauer** [giotto-ai/giotto-deep#44]
To accumulate class probabilities as in
https://github.com/giotto-ai/giotto-deep/blob/2965f49a5653bc51790bc76c66e21f2e5ca3e65d/gdeep/pipeline/pipeline.py#L110
could easily lead to `CUDA out of memory` problems.
In this case, it's not a problem because it's in the `with torch.no_grad()` scope and hence the gradients are not tracked.
But as pointed out in a different issue report on should use `model.eval()` and in this case, Cuda runs out of memory.

**In short: we should not accumulate tensors that require grad.**
See also _My model reports “cuda runtime error(2): out of memory” in
https://pytorch.org/docs/stable/notes/faq.html_
